### PR TITLE
[CI] Test latest artifacts of SF instead of the stable version

### LINKED ISSUE #11137: [ci] Coq should not be released with broken software foundations
IMHO that's a release blocker.

Software Foundations is one of the most popular Coq packages ever, we should not release a public beta with it being broken; it will cause a lot of people to lose a lot of time.

### PR DESCRIPTION
**Pull request by maximedenes**
Fixes #11137.

### CODE REVIEW

File: dev/ci/ci-sf.sh [hunk 1]
```diff
@@ -3,7 +3,11 @@
 ci_dir="$(dirname "$0")"
 . "${ci_dir}/ci-common.sh"
 
+CIRCLE_SF_TOKEN=00127070c10f5f09574b050e4f08e924764680d2
 mkdir -p "${CI_BUILD_DIR}" && cd "${CI_BUILD_DIR}" || exit 1
+sf_lf_CI_TARURL=`curl https://circleci.com/api/v1.1/project/gh/DeepSpec/sfdev/latest/artifacts?circle-token=${CIRCLE_SF_TOKEN} | jq '.[].url' | grep \/lf.tgz`
```
Comment (SkySkimmer) [inline dev/ci/ci-sf.sh]:
```suggestion
sf_lf_CI_TARURL=`curl https://circleci.com/api/v1.1/project/gh/DeepSpec/sfdev/latest/artifacts?circle-token=${CIRCLE_SF_TOKEN} | jq '.[] | select (.path == "lf.tgz") | .url'`
```
and so on for the others

**Comment by SkySkimmer** [comment]
Is the plan to test the SF dev version for 8.11 too? (ie what should be the milestone)
cc @ppedrot

**Comment by ppedrot** [comment]
I think so, let me put it into the appropriate milestone.

**Review by SkySkimmer — DISMISSED** [review]
SF should stop being allow_failure. Also see my previous comment (not required if you dislike the proposed change).

**Comment by Zimmi48** [comment]
Fails with:

```
 /builds/coq/coq/dev/ci/ci-sf.sh: line 8: curl: command not found
```

**Comment by SkySkimmer** [comment]
Pushed change.

**Comment by SkySkimmer** [comment]
Doesn't actually work, with curl I get not output when fetching the tgz, with wget I get `400 Bad Request`

**Comment by maximedenes** [comment]
> with wget I get `400 Bad Request`

Really? It works for me.

**Comment by SkySkimmer** [comment]
Maybe you have some login stuff on your system beyond the token?

**Comment by maximedenes** [comment]
> Maybe you have some login stuff on your system beyond the token?

It could be, but I have no idea how it works, then.

**Comment by maximedenes** [comment]
> Maybe you have some login stuff on your system beyond the token?

Also, shouldn't it be a 401 for you, then?

**Comment by SkySkimmer** [comment]
Real pipeline is https://gitlab.com/coq/coq/pipelines/98395393 btw, coqbot got confused as I had to do the manual dance since we need to add jq to docker.

**Comment by SkySkimmer** [comment]
Pipeline also got bad request.

File: dev/ci/ci-sf.sh [hunk 2]
```diff
@@ -3,10 +3,18 @@
...
+CIRCLE_SF_TOKEN=00127070c10f5f09574b050e4f08e924764680d2
+data=$(wget https://circleci.com/api/v1.1/project/gh/DeepSpec/sfdev/latest/artifacts?circle-token=${CIRCLE_SF_TOKEN} -O -)
+
+mkdir -p "${CI_BUILD_DIR}" && cd "${CI_BUILD_DIR}"
+
+sf_lf_CI_TARURL=$(echo "$data"  | jq -rc '.[] | select (.path == "lf.tgz") | .url')
+sf_plf_CI_TARURL=$(echo "$data" | jq -rc '.[] | select (.path == "plf.tgz") | .url')
+sf_vfa_CI_TARURL=$(echo "$data" | jq -rc '.[] | select (.path == "vfa.tgz") | .url')
+
+wget -O - "${sf_lf_CI_TARURL}" | tar xvz
```
Comment (ejgallego) [inline dev/ci/ci-sf.sh]: Indeed this step requires authentication, artifacts are not public so we will get a "must be logged in" message.

**Review by ejgallego — CHANGES_REQUESTED** [review]
Downloading artifacts from circle requires auth.

**Comment by Zimmi48** [comment]
A solution around this would be to download fresh artifacts only when run on a protected branch (with a secret) and to store these as artifacts in our builds, and reuse them in the PR builds.

**Comment by ejgallego** [comment]
> A solution around this would be to download fresh artifacts only when run on a protected branch (with a secret) and to store these as artifacts in our builds, and reuse them in the PR builds.

Yup, that's seems like the good option for short-term, but that is still quite messy; maybe we should drop SF from CI and move it to platform?

**Comment by ppedrot** [comment]
If this is not solved very soon, I'll have to postpone.

**Comment by ejgallego** [comment]
@maximedenes , could we maybe put a temporary tar somewhere and use that for the moment?

**Comment by ejgallego** [comment]
I've added authentication to the artifact download, at least locally that did the trick for me.

**Review by ejgallego — APPROVED** [review]
This seems to be working now!! I can merge myself tho maybe I should get approval from @SkySkimmer

**Review by SkySkimmer — APPROVED** [review]

**Pull request merged**